.github/workflows/scripts/generate-weekly-report.js

```diff
@@ -0,0 +1,428 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+const fs = require('fs');
+const path = require('path');
+const yaml = require('js-yaml');
+
+
+const REPO_NAME = "opentelemetry-collector-contrib"
+const REPO_OWNER = "open-telemetry"
+
+function debug(msg) {
+  console.log(JSON.stringify(msg, null, 2))
+}
+
+async function getIssues(github, queryParams, filterPrs = true) {
+  let allIssues = [];
+  try {
+    while (true) {
+      const response = await github.issues.listForRepo(queryParams);
+      // filter out pull requests
+      const issues = filterPrs ? response.data.filter(issue => !issue.pull_request) : response.data;
+      allIssues = allIssues.concat(issues);
+
+      // Check the 'link' header to see if there are more pages
+      const linkHeader = response.headers.link;
+      if (!linkHeader || !linkHeader.includes('rel="next"')) {
+        break;
+      }
+
+      queryParams.page++;
```

Comment:
What happens if a new issue is created during paging? Will the results be distorted, meaning one of the issues will be duplicated in the results? Is there a way to work around this?

Reply:
this won't happen because the report will only look at issues created since midnight the day before the action is run (to precisely avoid this case). see https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/26111/files#diff-6d378c7867e0f7f3eb58ea282fa1d63937c870c9e55825a4600d3f935592cbb5R73